pkg/components/fluentd/run_script.go

```diff
@@ -96,10 +96,26 @@ if type -p jemalloc-config > /dev/null 2>&1 ; then
     export LD_PRELOAD=$( jemalloc-config --libdir )/libjemalloc.so.$( jemalloc-config --revision )
     export LD_BIND_NOW=1 # workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1544815
 fi
-if [ -f /var/log/openshift-apiserver/audit.log.pos ] ; then
-  #https://bugzilla.redhat.com/show_bug.cgi?id=1867687
-  mv /var/log/openshift-apiserver/audit.log.pos /var/log/oauth-apiserver.audit.log
+
+# In case of an update to a secure fluentd container, copy the fluentd pos files to their new
+# locations under /var/lib/fluentd/pos. Moving old pos files is not possible since /var/log
+# is mounted read-only in the secure fluentd container.
+#
+if [ -f /var/log/openshift-apiserver/audit.log.pos -a ! -f /var/lib/fluentd/pos/oauth-apiserver.audit.log ] ; then
+    cp /var/log/openshift-apiserver/audit.log.pos /var/lib/fluentd/pos/oauth-apiserver.audit.log
 fi
+declare -A POS_FILES_FROM_TO=( [/var/log/audit/audit.log.pos]=/var/lib/fluentd/pos/audit.log.pos [/var/log/kube-apiserver/audit.log.pos]=/var/lib/fluentd/pos/kube-apiserver.audit.log.pos )
+for pos_file in es-containers.log.pos journal_pos.json oauth-apiserver.audit.log
+do
+  POS_FILES_FROM_TO["/var/log/$pos_file"]="/var/lib/fluentd/pos/$pos_file"
```

Comment:
Do we have to precreate the directory so the copy wont fail?

Reply:
Yes we do, thanks for catching.